torchvision/prototype/transforms/_augment.py

```diff
@@ -180,3 +184,190 @@ def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
+
+        # Copy-paste masks:
+        masks = masks * (~paste_alpha_mask)
+        non_all_zero_masks = masks.sum((-1, -2)) > 0
+        masks = masks[non_all_zero_masks]
+
+        # Do a shallow copy of the target dict
+        out_target = {k: v for k, v in target.items()}
+
+        out_target["masks"] = torch.cat([masks, paste_masks])
+
+        # Copy-paste boxes and labels
+        boxes = masks_to_boxes(masks)
+        out_target["boxes"] = torch.cat([boxes, paste_boxes])
+
+        labels = target["labels"][non_all_zero_masks]
+        out_target["labels"] = torch.cat([labels, paste_labels])
+
+        # Update additional optional keys: area and iscrowd if exist
+        if "area" in target:
+            out_target["area"] = out_target["masks"].sum((-1, -2)).to(torch.float32)
+
+        if "iscrowd" in target and "iscrowd" in paste_target:
+            # target['iscrowd'] size can be differ from mask size (non_all_zero_masks)
+            # For example, if previous transforms geometrically modifies masks/boxes/labels but
+            # does not update "iscrowd"
+            if len(target["iscrowd"]) == len(non_all_zero_masks):
+                iscrowd = target["iscrowd"][non_all_zero_masks]
+                paste_iscrowd = paste_target["iscrowd"][random_selection]
+                out_target["iscrowd"] = torch.cat([iscrowd, paste_iscrowd])
+
+        # Check for degenerated boxes and remove them
+        boxes = out_target["boxes"]
+        degenerate_boxes = boxes[:, 2:] <= boxes[:, :2]
+        if degenerate_boxes.any():
+            valid_targets = ~degenerate_boxes.any(dim=1)
+
+            out_target["boxes"] = boxes[valid_targets]
+            out_target["masks"] = out_target["masks"][valid_targets]
+            out_target["labels"] = out_target["labels"][valid_targets]
+
+            if "area" in out_target:
+                out_target["area"] = out_target["area"][valid_targets]
+            if "iscrowd" in out_target and len(out_target["iscrowd"]) == len(valid_targets):
+                out_target["iscrowd"] = out_target["iscrowd"][valid_targets]
+
+        return image, out_target
+
+    def forward(self, *inputs: Any) -> Any:
+        sample = inputs if len(inputs) > 1 else inputs[0]
+
+        flat_sample, spec = tree_flatten(sample)
+
+        # fetch all images, bboxes, masks and labels from unstructured input
+        # with List[image], List[BoundingBox], List[SegmentationMask], List[Label]
+        images, bboxes, masks, labels = [], [], [], []
+        for obj in flat_sample:
+            if isinstance(obj, PIL.Image.Image, features.Image) or is_simple_tensor(obj):
+                images.append(F.to_image_tensor(obj))
+            elif isinstance(obj, features.BoundingBox):
+                bboxes.append(obj)
+            elif isinstance(obj, features.SegmentationMask):
+                masks.append(obj)
+            elif isinstance(obj, (features.Label, features.OneHotLabel)):
+                labels.append(obj)
```

Comment:
As our input is a non-collated batch of datapoint, we need to fetch images, bboxes, masks, labels.
We then transform them and finally put into their places.
I'm not a fan of this code but this helps to do what we want. Maybe we could specify a bit more the input structure and thus code `forward` in a more elegant way.